app-modules/basic-needs/src/Filament/Resources/BasicNeedProgramResource/Pages/CreateBasicNeedProgram.php

```diff
@@ -0,0 +1,106 @@
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\BasicNeeds\Filament\Resources\BasicNeedProgramResource\Pages;
+
+use Filament\Forms\Form;
+use Laravel\Pennant\Feature;
+use Filament\Forms\Components\Select;
+use Filament\Forms\Components\Textarea;
+use Filament\Forms\Components\TextInput;
+use Filament\Resources\Pages\CreateRecord;
+use AdvisingApp\BasicNeeds\Models\BasicNeedCategory;
+use AdvisingApp\BasicNeeds\Filament\Resources\BasicNeedProgramResource;
+
+class CreateBasicNeedProgram extends CreateRecord
+{
+    protected static string $resource = BasicNeedProgramResource::class;
+
+    public function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+                TextInput::make('name')
+                    ->label('Program Name')
+                    ->required()
+                    ->string()
+                    ->maxLength(255)
+                    ->unique(),
+                Textarea::make('description')
+                    ->label('Description')
+                    ->maxLength(65535)
+                    ->string(),
+                Select::make('basic_need_category_id')
+                    ->label('Program Category')
+                    ->required()
+                    ->relationship('basicNeedCategories', 'name')
+                    ->visible(Feature::active('basic-needs-category'))
+                    ->native(false)
+                    ->searchable()
+                    ->getSearchResultsUsing(fn (string $search): array => BasicNeedCategory::where('name', 'like', "%{$search}%")->limit(50)->pluck('name', 'id')->toArray())
```

Comment:
Is this needed? I believe the relationship on the select should take care of making sure this works this way